### ISSUE TITLE
Version cookie pollution

### ISSUE THREAD

**Issue by joaquintides** [boostorg/website-v2#1168]
Suppose a user has their version cookie set to 1.85, so the URL

https://www.boost.io/releases/

gets resolved to

https://www.boost.io/releases/boost-1-85-0/

as it should be. Now, if the user manually inserts something like this on their URL bar:

https://www.boost.io/releases/whatever

The site returns a 404 (ok) but then clicking on "Releases" or "Libraries" fails with 404 or 500 because the version cookie has been changed to "whatever".

In short, the version cookie must be validated before its value is changed.

**Comment by kennethreitz** [comment]
will look into this, thanks
